test cases/common/251 add_project_dependencies/meson.build

```diff
@@ -0,0 +1,19 @@
+project('zlib system dependency', 'c')
+
+cc = meson.get_compiler('c')
+
+z = dependency('zlib', method: 'system', required: false)
+if not z.found()
+  error('MESON_SKIP_TEST zlib not present')
+endif
```

Comment:
I actually would like to see in this unit test  adding `m_dep = cc.find_library('m')` to validate it works with library dependencies and also with not-found dependencies. From the code I'm pretty sure it works, but just to be sure.

Reply:
Done.